src/graph.rs

```diff
@@ -246,22 +246,26 @@ impl Graph {
     ///
     /// Returns an error if:
     ///   * `output` is not in `graph`.
+    ///   * `num_dims` does not match the actual number of dimensions.
```

Comment:
This should probably be removed.  If this happens, it's an internal error in our library.  I don't think users should have to worry about it.